script.js

```diff
@@ -1,12 +1,284 @@
+  rootElem.append(pageHeaderElement);
+}
+
+function createShowsDropDownMenu() {
+  let dropDownMenuElement = document.createElement("select");
+  dropDownMenuElement.id = "show-select";
+  let defaultOption = document.createElement("option");
+  defaultOption.innerText = "Select Show";
+  dropDownMenuElement.append(defaultOption);
+
+  let allShows = getAllShows();
+
+  // good example of where to use a ternirary
+  allShows.sort(function (a, b) {
+    // if (a.name < b.name) {
+    //   return -1;
+    // }
+    // else {
+    //   return 1;
+    // }
+
+    return a.name < b.name ? 1 : -1;
+  });
+
+  for (const show of allShows) {
+    let dropDownOption = document.createElement("option");
+    //////////// Assign dropdown option values
+    dropDownOption.value = show.id;
+
+    dropDownOption.innerText = `${show.name}`;
+    dropDownMenuElement.append(dropDownOption);
+  }
+
+  let header = document.getElementById("page-header");
+  header.append(dropDownMenuElement);
+}
+
+function loadNewShow() {
+  let showDropDownMenu = document.getElementById("show-select");
+  let showDropDownChoice = showDropDownMenu.value;
+
+  showDropDownMenu.addEventListener("change", populateNewEpisodes);
+
+  function populateNewEpisodes() {
+    showDropDownChoice = showDropDownMenu.value;
+    let showURL = `https://api.tvmaze.com/shows/${showDropDownChoice}/episodes`;
+
+    // before making cards for new show, clear html
+    document.querySelector("#root").innerHTML = "";
+    getAllEpisodesFromAPI(showURL).then(makePageForEpisodes);
+  }
+}
+
+function createDropDownMenu(episodeList) {
+  // make a dropdown menu
+  // for loop on episode list to populate options in dropdown
+
+  let dropDownMenuElement = document.createElement("select");
+  dropDownMenuElement.id = "episode-select";
+  let defaultOption = document.createElement("option");
+  defaultOption.innerText = "Select Episode";
+  dropDownMenuElement.append(defaultOption);
+
+  for (const episode of episodeList) {
+    let dropDownOption = document.createElement("option");
+    //////////// Assign dropdown option values
+    dropDownOption.value = "EP" + episode.id;
+
+    // more padstart stuff -- let's put this in a function later
+    const paddedSeasonNumber = episode.season.toString().padStart(2, "0");
+    const paddedEpisodeNumber = episode.number.toString().padStart(2, "0");
+    const episodeName = episode.name;
+
+    dropDownOption.innerText = `S${paddedSeasonNumber}E${paddedEpisodeNumber} - ${episodeName}`;
+    dropDownMenuElement.append(dropDownOption);
+  }
+
+  let header = document.getElementById("page-header");
+  header.append(dropDownMenuElement);
+}
+
+function useDropdownToJumpToEpisode(episodeList) {
+  let episodeDropDownMenu = document.getElementById("episode-select");
+  let episodeDropDownChoice = episodeDropDownMenu.value;
+
+  episodeDropDownMenu.addEventListener("change", jumpToEpisode);
+
+  function jumpToEpisode() {
+    for (const episode of episodeList) {
```

Comment:
It would be simpler and quicker here, rather than looping through all the episodes, to just find the card element whose `id` value is the value of the dropdown - assuming it is non-empty of course! The use of `.scrollIntoView()` here is really good.